### PR TITLE
feat: make user to use existing Viewport and Camera element

### LINKED ISSUE #198: If Viewport or Camera element exists, use it.
## Description
Originally, Flicking generates its own Viewport and Camera DOM element to use.
But, in some case like using in framework-based environment like React, Vue, Angular.
It is more natural to have Viewport and Camera element to be generated by framework. 

## Spec
If element with class named {user-prefix}-flick-viewport, and {user-prefix}-flick-camera exists, use them for Viewport and Camera element.

### PR DESCRIPTION
**Pull request by WoodNeck**
## Issue
#198

## Details
- This resolves #198 
  - It also can cope with `destroy()` calls.
- It includes some refactoring, and also adds some tests for new feature.

### CODE REVIEW

File: src/components/Viewport.ts [hunk 1]
```diff
@@ -473,16 +482,35 @@ export default class Viewport {
...
   public destroy(): void {
+    const state = this.state;
+    const wrapper = this.flicking.getElement();
     const viewportElement = this.viewportElement;
-    const wrapper = viewportElement.parentElement;
+    const cameraElement = this.cameraElement;
+
+    const originalPanels = this.panelManager.originalPanels();
+
+    restoreStyle(viewportElement, state.originalViewportStyle);
```
Comment (daybrush) [inline src/components/Viewport.ts]: @WoodNeck 
Do you think the condition to call restoreStyle function also requires `isCameraGiven` and `isViewportGiven`?

File: src/components/Viewport.ts [hunk 2]
```diff
@@ -700,16 +729,78 @@ export default class Viewport {
...
+    const viewportCandidate = wrapper.children[0] as HTMLElement;
+    const hasViewportElement = hasClass(viewportCandidate, `${classPrefix}-viewport`);
+
+    const viewportElement = hasViewportElement
+      ? viewportCandidate
+      : document.createElement("div");
+
+    const cameraCandidate = hasViewportElement
+      ? viewportElement.children[0] as HTMLElement
+      : wrapper.children[0] as HTMLElement;
```
Comment (daybrush) [inline src/components/Viewport.ts]: @WoodNeck 
Is there a camera under the wrapper?

File: src/components/Viewport.ts (hunk 1 above)
Comment (WoodNeck) [reply src/components/Viewport.ts]: Well that would be more efficient, but I think that's still trivial.

File: src/components/Viewport.ts (hunk 2 above)
Comment (WoodNeck) [reply src/components/Viewport.ts]: I've considered all 4 possibilities.
Which is,
1) No Viewport, No Camera element
2) Viewport element exists, while no camera element.
3) Camera element exists, while no viewport element.
4) Both viewport and camera element exist.

File: test/unit/initialize.spec.ts [hunk 3]
```diff
@@ -555,7 +609,10 @@ describe("Initialization", () => {
             const endIndex = flicking.getIndex();
             const indexAtDestPos = Math.floor(destPos / panelWidth);
             // As all delta is above threshold, it should change panel at least once
-            const expectedIndex = Math.min(nearestPanel.getIndex() + snapCount, Math.max(1, indexAtDestPos));
+            const expectedIndex = destPos >= 100
+              ? Math.min(nearestPanel.getIndex() + snapCount, Math.max(1, indexAtDestPos))
+              // Sometimes, simulation won't return proper dest position for given delta.
```
Comment (jongmoon) [inline test/unit/initialize.spec.ts]: I suggest to prepend `TODO:` to check later. I think this code may have potential problem.

Comment (WoodNeck) [reply test/unit/initialize.spec.ts]: It's been changed at other PR(#203). That's quite more detailed, so please check it after I rebase it.

**Pull request merged**